packages/mobile/src/store/offline-downloads/sagas/migrateOfflineDataPathSaga.ts

```diff
@@ -0,0 +1,63 @@
+import path from 'path'
+
+import RNFS from 'react-native-fs'
+import RNFetchBlob from 'rn-fetch-blob'
+import { call } from 'typed-redux-saga'
+
+import { downloadsRoot } from 'app/services/offline-downloader'
+
+import { getIsOfflineEnabled } from './getIsOfflineEnabled'
+
+const {
+  fs: { dirs, unlink, exists }
+} = RNFetchBlob
+
+// Current migration: 4/3/2023
+// dirs.DocumentDir -> dirs.CacheDir
+const legacyDownloadsRoot = path.join(dirs.DocumentDir, 'downloads')
+
+// Move downloads from legacy storage location to the updated path (currently: /Documents -> /Caches)
+export function* migrateOfflineDataPathSaga() {
+  const isOfflineModeEnabled = yield* call(getIsOfflineEnabled)
+  if (!isOfflineModeEnabled) return
+
+  const legacyFilesExist = yield* call(exists, legacyDownloadsRoot)
+  if (!legacyFilesExist) return // TODO: fire action?
+
+  try {
+    yield* call(copyRecursive, legacyDownloadsRoot, downloadsRoot)
+  } finally {
```

Comment:
awesome, so what do we do with the redux-persist data? if we fail, maybe we want to update status for all items?

Reply:
Yeah I was thinking about that too. Probably a good idea to put them all back in queue again?

Reply:
yeah that makes sense... do we even want to remove, or just keep those around indefinitely? just note that we tried and those will remain indefinitely? basically trying to prevent user impact. but if we think failure is unlikely, then maybe we dont need to worry about the whole thing? just main thing is to make sure the states stay in-sync, whether that's full deleted, or migrated?

Reply:
Two rules I'd like to keep:
1. I think it's best if we remove all trace of downloaded files visible in the browser.
2. I'd like to keep it such that nobody has files in both places unless the transfer is currently in progress.

With that in mind, when we fail, I want to treat the destination path as the source of truth and update state to match. In theory, since you should only need the migration on first load, the destination should be empty, so if we fail, we could:
1. Wipe both dirs and throw everything marked for download back into the queue
2. Hope that some of it got there and make sure sync is up to the task of figuring out what files are missing. Notably, I don't think it will do that correctly at the moment for tracks until the staleness window has passed.

I think the cleanest approach will be to wipe everything. This is a fail case anyway, so I don't necessarily mind incurring the additional cost.

Reply:
awesome that makes a ton of sense, lets go with wiping everything in a failure case